File: snippets/fsharp/VS_Snippets_CLR_System/system.Func~1/fs/Func1.fs
module Func1

// <Snippet2>
open System
open System.IO

type OutputTarget() =
    member _.SendToFile() =
        try
            let fn = Path.GetTempFileName()
            use sw = new StreamWriter(fn)
            sw.WriteLine "Hello, World!"
            true
        with _ ->
            false

let output = OutputTarget()
let methodCall = Func<bool> output.SendToFile
if methodCall.Invoke() then
    printfn "Success!"
else
    printfn "File write operation failed."

// </Snippet2>

File: snippets/csharp/System/FuncTResult/Overview/Func1.cs
// <Snippet2>
using System;
using System.IO;

public class FuncDelegateExample
{
    public static void Run()
    {
        FuncDelegateOutputTarget output = new();
        Func<bool> methodCall = output.SendToFile;
        if (methodCall())
            Console.WriteLine("Success!");
        else
            Console.WriteLine("File write operation failed.");
    }
}

public class FuncDelegateOutputTarget
{
    public bool SendToFile()
    {
        try
        {
            string fn = Path.GetTempFileName();
            using StreamWriter sw = new(fn);
            sw.WriteLine("Hello, World!");
            return true;
        }
        catch
        {
            return false;
        }
    }
}
// </Snippet2>
